internal/controllers/machinedeployment/machinedeployment_status.go

```diff
@@ -0,0 +1,502 @@
+	if machinesToBeRemediated == nil || unhealthyMachines == nil {
+		v1beta2conditions.Set(machineDeployment, metav1.Condition{
+			Type:    clusterv1.MachineDeploymentRemediatingV1Beta2Condition,
+			Status:  metav1.ConditionUnknown,
+			Reason:  clusterv1.MachineDeploymentRemediatingInternalErrorV1Beta2Reason,
+			Message: "Please check controller logs for errors",
+		})
+		return
+	}
+
+	if len(machinesToBeRemediated) == 0 {
+		message := aggregateUnhealthyMachines(unhealthyMachines)
+		v1beta2conditions.Set(machineDeployment, metav1.Condition{
+			Type:    clusterv1.MachineDeploymentRemediatingV1Beta2Condition,
+			Status:  metav1.ConditionFalse,
+			Reason:  clusterv1.MachineDeploymentNotRemediatingV1Beta2Reason,
+			Message: message,
+		})
+		return
+	}
+
+	remediatingCondition, err := v1beta2conditions.NewAggregateCondition(
+		machinesToBeRemediated.UnsortedList(), clusterv1.MachineOwnerRemediatedV1Beta2Condition,
+		v1beta2conditions.TargetConditionType(clusterv1.MachineDeploymentRemediatingV1Beta2Condition),
+	)
+	if err != nil {
+		v1beta2conditions.Set(machineDeployment, metav1.Condition{
+			Type:    clusterv1.MachineDeploymentRemediatingV1Beta2Condition,
+			Status:  metav1.ConditionUnknown,
+			Reason:  clusterv1.MachineDeploymentRemediatingInternalErrorV1Beta2Reason,
+			Message: "Please check controller logs for errors",
+		})
+
+		log := ctrl.LoggerFrom(ctx)
+		log.Error(err, fmt.Sprintf("Failed to aggregate Machine's %s conditions", clusterv1.MachineOwnerRemediatedV1Beta2Condition))
+		return
+	}
+
+	v1beta2conditions.Set(machineDeployment, metav1.Condition{
+		Type:    remediatingCondition.Type,
+		Status:  metav1.ConditionTrue,
+		Reason:  clusterv1.MachineDeploymentRemediatingV1Beta2Reason,
+		Message: remediatingCondition.Message,
+	})
+}
+
+func setDeletingCondition(_ context.Context, machineDeployment *clusterv1.MachineDeployment, machineSets []*clusterv1.MachineSet, machines collections.Machines, getAndAdoptMachineSetsForDeploymentSucceeded bool) {
+	// If we got unexpected errors in listing the machines sets or machines (this should never happen), surface them.
+	if !getAndAdoptMachineSetsForDeploymentSucceeded || machines == nil {
+		v1beta2conditions.Set(machineDeployment, metav1.Condition{
+			Type:    clusterv1.MachineDeploymentDeletingV1Beta2Condition,
+			Status:  metav1.ConditionUnknown,
+			Reason:  clusterv1.MachineDeploymentDeletingInternalErrorV1Beta2Reason,
+			Message: "Please check controller logs for errors",
+		})
+		return
+	}
+
+	if machineDeployment.DeletionTimestamp.IsZero() {
+		v1beta2conditions.Set(machineDeployment, metav1.Condition{
+			Type:   clusterv1.MachineDeploymentDeletingV1Beta2Condition,
+			Status: metav1.ConditionFalse,
+			Reason: clusterv1.MachineDeploymentDeletingDeletionTimestampNotSetV1Beta2Reason,
+		})
+		return
+	}
+
+	message := ""
+	if len(machines) > 0 {
+		message = fmt.Sprintf("Deleting %d replicas", len(machines))
```

Comment:
```suggestion
		message = fmt.Sprintf("Deleting %d Machines", len(machines))
```
nit. I know it's pretty much the same, but I still think Machines is clearer/better